### PR TITLE
Support for AVS 3.6

### PR DESCRIPTION
**Pull request by typfel**
Update calling implementation for API changes in AVS 3.6.

in short:
- call config retrieval has moved from the flow manager to the`CallingRequestStrategy`
- new API for multi account support

### CODE REVIEW

File: Cartfile.resolved [hunk 1]
```diff
@@ -13,6 +13,6 @@ github "wireapp/wire-ios-mocktransport" "49.1.0"
 github "wireapp/wire-ios-protos" "11.0.1"
 github "wireapp/wire-ios-request-strategy" "55.0.0"
 github "wireapp/wire-ios-system" "18.2.2"
-github "wireapp/wire-ios-testing" "12.0.0"
```
Comment (vytis) [inline Cartfile.resolved]: let's revert to `12.0.0`, test suite crashes because of leak check currently...

File: Source/Calling/AVSWrapper.swift [hunk 2]
```diff
@@ -23,106 +23,122 @@ import avs
...
             let serverTime = UInt32(callEvent.serverTimestamp.timeIntervalSince1970)
             
-            wcall_recv_msg(bytes, callEvent.data.count, currentTime, serverTime, callEvent.conversationId.transportString(), callEvent.userId.transportString(), callEvent.clientId)
+            wcall_recv_msg(handle, bytes, callEvent.data.count, currentTime, serverTime, callEvent.conversationId.transportString(), callEvent.userId.transportString(), callEvent.clientId)
+        }
+    }
+    
+    public func update(callConfig: String?, httpStatusCode: Int) {
+        if let callConfig = callConfig {
+            wcall_config_update(handle, httpStatusCode == 200 ? 0 : EPROTO, callConfig)
```
Comment (vytis) [inline Source/Calling/AVSWrapper.swift]: I guess if let is not really needed here as the else part simply passes nil

**Comment by vytis** [comment]
So we will still keep `V3` suffixes in class names?

File: Source/Calling/AVSWrapper.swift (hunk 2 above)
Comment (typfel) [reply Source/Calling/AVSWrapper.swift]: True I will revisit this part. But there's was issue passing an optional String into the C-function.

**Comment by typfel** [comment]
@vytis I'll remove the V3 prefixes when I have time to finish https://github.com/wireapp/wire-ios-sync-engine/pull/498

**Comment by vytis** [comment]
@typfel ah true, got confused which PR does what.

**Pull request merged**